axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/LegacyAxonServerEventStorageEngine.java

```diff
@@ -137,11 +115,12 @@ public CompletableFuture<AppendTransaction> appendEvents(@Nonnull AppendConditio
                 String aggregateIdentifier = resolveAggregateIdentifier(taggedEvent.tags());
                 String aggregateType = resolveAggregateType(taggedEvent.tags());
                 if (aggregateIdentifier != null && aggregateType != null && !taggedEvent.tags().isEmpty()) {
-                    long nextSequence = resolveSequencer(aggregateSequences, aggregateIdentifier, consistencyMarker).incrementAndGet();
+                    long nextSequence = aggregateSequencer.resolveBy(aggregateIdentifier).incrementAndGet();
                     builder.setAggregateIdentifier(aggregateIdentifier).setAggregateType(aggregateType)
                            .setAggregateSequenceNumber(nextSequence);
                 }
-                buildMetaData(event.getMetaData(), builder.getMetaDataMap());
+                var modifiableMetaDataMap = new HashMap<>(builder.getMetaDataMap());
+                buildMetaData(event.getMetaData(), modifiableMetaDataMap);
```

Comment:
There was a bug, because builder.getMetaDataMap() is Unmodifiable, so it was impossible to put metadata into Events.